source/common/stats/thread_local_store.cc

```diff
@@ -967,6 +961,16 @@ void ThreadLocalStoreImpl::forEachTextReadout(SizeFn f_size, StatFn<TextReadout>
   alloc_.forEachTextReadout(f_size, f_stat);
 }
 
+void ThreadLocalStoreImpl::forEachHistogram(SizeFn f_size, StatFn<ParentHistogram> f_stat) const {
+  Thread::LockGuard lock(hist_mutex_);
+  if (f_size != nullptr) {
+    f_size(histogram_set_.size());
+  }
+  for (auto histogram : histogram_set_) {
```

Comment:
`auto& histogram` to ensure it's a reference not a copy? I'm pretty sure the base type is a raw pointer so this won't be any different, but it's clearer reading it that we're not copying something expensive on accident.

Reply:
Oh good point, but I have an alternative suggestion as a remedy: use an explicit type, since it's not obvious from context what the exact type is. Then it will be clear it's a pointer, so there's no copy overhead.

@ggreenway WDYT?

Reply:
Fixed per Josh's suggestion.